### PR TITLE
Added GET requests to the front end

### LINKED ISSUE #1: Build Webapp skeleton
TODO: Define the requirements of the webapp interface

### LINKED ISSUE #2: Site Search Function
Site should allow a user to search for the ecolabel they are looking for.
Site will call to the backend server to get a list of potential match.
TODO: Define the structure of the call and return.

### LINKED ISSUE #3: EcoLabel Selector
After a search, the user should be able to select a specific label that they see on their product. The site will alert the server of the choice. The site might show the label itself, its name, and any other details that might help the user identify it.

### LINKED ISSUE #4: Chat Environment
The Chat Environment should initialize after the user selects an EcoLabel, and update when the Server serves a chat token, and the AI's initial response.
The Chat Env. Should allow the user to read the AI's messages, write their own messages, and send them. 

TODO: Define the structure of initial message, following messages, and the user message.

Sending a message will make a call including the chat token and message to the server, which will serve the response back.

Closing the Chat Env. will make a call to the server indicating the termination of the chat.

### PR DESCRIPTION
**Pull request by elsaschutfort**

### CODE REVIEW

File: app/ecolabel-translator/src/App.js [hunk 1]
```diff
@@ -9,14 +9,33 @@ function App() {
...
-  const handleLabelSubmit = (event) => {
+  const handleLabelSubmit = async (event) => {
     event.preventDefault();
-    setResults({
-      description: 'This is a sample eco-label description.',
-      epaRecommendation: 'EPA recommends this label for sustainable products.'
-    });
+  
+    try {
+      const response = await fetch(`http://127.0.0.1:5000/search?q=${label}`, {
```
Comment (ETGrif) [inline app/ecolabel-translator/src/App.js]: Can we get this into a variable somewhere? Then it can be easily changed when we move to prod

Comment (ETGrif) [inline app/ecolabel-translator/src/App.js]: When I ran the site, I couldn't get it to run. There is a type error in the fetch..
`Error submitting label: TypeError: Failed to fetch
    at handleLabelSubmit (App.js:18:1)`

I verified that the url used does actually access the API, so its not a connection error. It is also not a 422, so it wasn't thrown by the api

File: app/ecolabel-translator/src/App.js [hunk 2]
```diff
@@ -26,12 +45,26 @@ function App() {
...
   };
 
-  const handleMessageSubmit = (event) => {
+  const handleMessageSubmit = async (event) => {
     event.preventDefault();
-    if (input.trim()) {
-      setMessages([...messages, { sender: 'User', text: input }]);
+
+    try {
+      const response = await fetch(`http://127.0.0.1:5000/chat/send?t=${input}&m=${token}`, {
```
Comment (ETGrif) [inline app/ecolabel-translator/src/App.js]: Variable here too plz

File: app/ecolabel-translator/src/App.js [hunk 3]
```diff
@@ -40,6 +73,27 @@ function App() {
     console.log('Chat ended.');
   };
 
+  const handleChatActive = async () => {
+    setIsChatActive(true);
+    try {
+      const response = await fetch(`http://127.0.0.1:5000/chat/init?label=true`, {
```
Comment (ETGrif) [inline app/ecolabel-translator/src/App.js]: Variable, and also why ```?label=true```? chat/init label is the name of the label chosen by the user

**Review by ETGrif — CHANGES_REQUESTED** [review]
Doing good, but needs a little rework. I merged the main into this branch so that you can run the site alongside the server.
The api url should also be stored in a global, so that we can change it easier.
I also noticed that there wasn't a chat/terminate call! (or at least it didn't show up in the diff. I just want to make sure that stays in there.

File: app/ecolabel-translator/src/App.js (hunk 3 above)
Comment (elsaschutfort) [reply app/ecolabel-translator/src/App.js]: It's named label because that is what chat_init on the backend gets from the request.

Comment (ETGrif) [reply app/ecolabel-translator/src/App.js]: I'm concerned about the ```=true``` bit

**Comment by ETGrif** [comment]
Closes #1, closes #2, closes #3, closes #4

**Review by ETGrif — APPROVED** [review]
There still isn't a call to terminate the chat when we exit, but that can be a later issue, or at this point I don't care that much

**Pull request merged**